WalletWasabi/WabiSabi/Client/ArenaClient.cs

```diff
@@ -94,5 +94,44 @@ public async Task ConfirmConnectionAsync(Guid roundId, Guid aliceId, IEnumerable
 			AmountCredentialClient.HandleResponse(confirmConnectionResponse.ZeroAmountCredentials, zeroAmountCredentialResponseValidation);
 			WeightCredentialClient.HandleResponse(confirmConnectionResponse.ZeroWeightCredentials, zeroWeightCredentialResponseValidation);
 		}
+
+		public async Task SignTransactionAsync(Guid roundId, IEnumerable<ICoin> coinsToSign, BitcoinSecret bitcoinSecret, Transaction unsignedCoinJoin)
+		{
+			if (unsignedCoinJoin.Inputs.Count == 0)
+			{
+				throw new ArgumentException("No inputs to sign.", nameof(unsignedCoinJoin));
+			}
+
+			if (!coinsToSign.Any())
+			{
+				throw new ArgumentException("No coins were provided.", nameof(coinsToSign));
+			}
+
+			var myInputs = coinsToSign.ToDictionary(c => c.Outpoint);
+			var signedCoinJoin = unsignedCoinJoin.Clone();
```

Comment:
Just a nitpick: should the variable name be `clonedUnsignedCoinJoin` or `coinJoinToSign?

Another question: why clone this, why not simply use the `unsignedCoinJoin` variable?

Reply:
The transaction to sign should not be mutated as a side effect of producing and submitting the signature as that object belongs to the caller.

Cloning in this way when operating on mutable types is a desirable convention because side effects in general are harder to reason about and the name should indicate that things are being modified. In the tests for example the transaction is just shared, introducing the side effect of mutating the transaction to be signed would make the them incorrect, since if the client code would end up making the test pass even if the server did not update the round transaction (easy to fix by just not sharing the transaction, but this just illustrates the kind of things you need to start worrying about when code mutates state that it doesn't clearly own)

Reply:
Both names has its rationale, so I keep the original one.